rust/hyperlane-base/src/contract_sync/cursor.rs

```diff
@@ -518,17 +511,19 @@ where
         };
 
         let rate_limit = self.get_rate_limit().await?;
-        let action = if let Some(rate_limit) = rate_limit {
-            CursorAction::Sleep(rate_limit)
-        } else {
-            // According to the logic in `get_rate_limit` we should be able to safely unwrap here.
-            // However, take the safer option and fall back to a zero duration sleep
-            match self.sync_state.get_next_range(None, None).await? {
-                Some(range) => CursorAction::Query(range),
-                None => CursorAction::Sleep(Duration::from_secs(0)),
+        if let Some(rate_limit) = rate_limit {
+            return Ok((CursorAction::Sleep(rate_limit), eta));
+        }
+        let (payment_count, tip) = self.indexer.sequence_at_tip().await?;
+        let cursor_count = self.sync_state.next_sequence;
+        if cursor_count < payment_count {
+            if let Some(range) = self.sync_state.get_next_range(payment_count, tip).await? {
+                return Ok((CursorAction::Query(range), eta));
             }
-        };
-        Ok((action, eta))
+        }
+
+        // TODO: Define the sleep time from interval flag
```

Comment:
Should we do this in this PR?

Reply:
TBH I copied this comment from [another line](https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/dce883d3a19b9727196c9d27db06d2e50f60850b/rust/hyperlane-base/src/contract_sync/cursor.rs#L415) in this file. The interval flag is only present in the validator but these cursors are only used by the relayer and scraper I think? So we'd actually need to add the interval to the relayer settings struct